src/web/assets/cp/src/js/BaseElementSelectorModal.js

```diff
@@ -24,17 +24,19 @@ Craft.BaseElementSelectorModal = Garnish.Modal.extend({
     init: function(elementType, settings) {
         this.elementType = elementType;
         this.setSettings(settings, Craft.BaseElementSelectorModal.defaults);
+        var $headingId = 'elementSelectorModalHeading-' + Date.now();
 
         // Build the modal
-        var $container = $('<div class="modal elementselectormodal"></div>').appendTo(Garnish.$bod),
+        var $container = $('<div class="modal elementselectormodal" aria-labelledby="' + $headingId + '"></div>').appendTo(Garnish.$bod),
+            $heading = $('<h2 id="' + $headingId + '" class="visually-hidden">' + this.settings.name + '</h2>').appendTo($container),
```

Comment:
@brandonkelly ideally the heading of each modal would reflect the field name (i.e. "Select featured image", "Select categories", etc.). I was unsure how to get the specific field names here, so I added a generic translation for the time being.

Reply:
@brandonkelly I've added two new translations 'Select element' and 'Select {element}' to handle this. The latter pulls the name from the fieldset legend node. Let me know if you prefer a different naming convention for these